src/azul/service/hca_response_v5.py

```diff
@@ -388,28 +385,30 @@ def make_organoid(self, organoid):
     def make_organoids(self, entry):
         return [self.make_organoid(organoid) for organoid in entry["contents"]["organoids"]]
 
-    def make_sample(self, sample):
-        lookup = {
-            'cell_lines': ('cellLines', self.make_cell_line),
-            'organoids': ('organoids', self.make_organoid),
-            'specimens': ('specimens', self.make_specimen),
-        }
-        entity_type = sample['entity_type']
-        if isinstance(entity_type, list):
-            entity_type, make_functions = map(list, zip(*map(lookup.get, entity_type)))
-            entity_dicts = (make_function(sample) for make_function in make_functions)
-            entity_dict = ChainMap(*entity_dicts)
-        else:
-            entity_type, make_function = lookup[entity_type]
-            entity_dict = make_function(sample)
+    def make_sample(self, sample, entity_dict, entity_type):
+        is_aggregate = isinstance(sample['document_id'], list)
+        organ_prop = 'organ' if entity_type == 'specimens' else 'model_organ'
         return {
-            'sampleEntityType': entity_type,
-            'effectiveOrgan': sample['effective_organ'],
+            'sampleEntityType': [entity_type] if is_aggregate else entity_type,
+            'effectiveOrgan': sample[organ_prop],
             **entity_dict
         }
 
     def make_samples(self, entry):
-        return [self.make_sample(sample) for sample in entry["contents"]["samples"]]
+        return [
```

Comment:
I think this can be compressed with a comprehension over a structure like `[(self.make_cell_line, 'cellLines', 'cell_lines'), …]`